packages/react-router/lib/hooks.tsx

```diff
@@ -100,7 +100,7 @@ export function useInRouterContext(): boolean {
  *
  * @see https://reactrouter.com/hooks/use-location
  */
-export function useLocation(): Location {
+export function useLocation<T = unknown>(): Location<T | null> {
```

Comment:
Should we default it to `any` instead of `unknown`? 
Because then the tsc won't complain about the state's type.